gen/property.go

```diff
@@ -0,0 +1,53 @@
+package gen
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+const tagName = "property"
+
+func Classify(group interface{}) error {
+	v := reflect.ValueOf(group)
+	if v.Kind() != reflect.Ptr || v.IsNil() {
+		return fmt.Errorf("TODO: Some error about invalid type")
+	}
+
+	e := v.Elem()
+	t := e.Type()
+
+	var raws []json.RawMessage
+	properties := make(map[string]reflect.Value)
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		tag := field.Tag.Get(tagName)
+		if tag == "RAW" {
+			raws = e.Field(i).Interface().([]json.RawMessage)
+		} else {
+			properties[tag] = e.Field(i)
+		}
+	}
+
+	// TODO: return err if there is no raw?
+
+	for _, raw := range raws {
+		var prop struct {
+			Property string
+		}
+		if err := json.Unmarshal(raw, &prop); err != nil {
+			return err
+		}
+		tgt := properties[prop.Property]
```

Comment:
TODO (only bother if the idea of `Classify` is deemed acceptable): these names (tgt, z, zi, zv) all are terrible, make better ones